### PR TITLE
[KAF-227] suppress/revive support

### PR DESCRIPTION
**Pull request by mgummelt**
To support suppress/revive 

Also adds ObjectPropertyDeserializer to support reading serialized java objects from zk

Adds SchedulerState (factored out from KafkaStateStore)

### CODE REVIEW

**Comment by mgummelt** [comment]
@gabrielhartmann Needs review

File: src/main/java/org/apache/mesos/curator/CuratorStateStore.java [hunk 1]
```diff
@@ -354,6 +355,35 @@ public void storeProperty(final String key, final byte[] value) throws StateStor
         }
     }
 
+    public void storePropertyAsObj(final String key, final Object obj) throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/curator/CuratorStateStore.java]: Should these be part of the `StateStore` interface?

File: src/main/java/org/apache/mesos/offer/ResourceUtils.java [hunk 2]
```diff
@@ -138,7 +138,7 @@ public static Resource getUnreservedScalar(String name, double value) {
 
     public static Resource getExpectedScalar(
             String name,
-            double value,
+           double value,
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/offer/ResourceUtils.java]: Accidental deletion?

File: src/main/java/org/apache/mesos/reconciliation/Reconciler.java [hunk 3]
```diff
@@ -12,8 +12,7 @@
 public interface Reconciler {
 
     /**
-     * Starts reconciliation against the provided tasks, which should represent what the Scheduler
-     * currently knows about task status.
+     * Starts reconciliation.
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/reconciliation/Reconciler.java]: `Starts Task reconciliation`

File: src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java [hunk 4]
```diff
@@ -85,7 +98,8 @@ public DefaultRecoveryScheduler(
             recoveryRequirement = Optional.of(recoveryCandidates.get(new Random().nextInt(recoveryCandidates.size())));
         }
 
-        if (recoveryRequirement.isPresent() && launchConstrainer.canLaunch(recoveryRequirement.get())) {
+        boolean b = recoveryRequirement.isPresent() && launchConstrainer.canLaunch(recoveryRequirement.get());
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java]: Why does the `b` variable exist?  Shouldn't this all be part of the if statement below?

File: src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java [hunk 5]
```diff
@@ -127,11 +148,13 @@ public DefaultRecoveryScheduler(
...
     }
 
-    private void updateRecoveryPools(Collection<TaskInfo> terminatedTasks) {
+    private void updateRecoveryStatus(Collection<TaskInfo> terminatedTasks) {
         List<TaskInfo> failed = new ArrayList<>(terminatedTasks.stream()
                 .filter(failureMonitor::hasFailed)
                 .collect(Collectors.toList()));
         failed = failed.stream().distinct().collect(Collectors.toList());
+
+        // Why? They're already stored as failed.
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java]: File a JIRA or a TODO, please.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState extends CuratorStateStore implements TaskStatusProvider {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Prefer composition over inheritance.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+    private static final String SUPPRESSED_KEY = "suppressed";
+
+    public SchedulerState(String frameworkName, String mesosZkURI) {
+        super(frameworkName, mesosZkURI);
+    }
+
+    public FrameworkID getFrameworkId() {
+        try {
+            return fetchFrameworkId().orElse(null);
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: We just went through a series of changes to avoid returning nulls.  Let's not transform [the underlying `StateStore` interface](https://github.com/mesosphere/dcos-commons/blob/6a5e665c56dcb9eb478b89619453f4f5e0fdbad2/src/main/java/org/apache/mesos/state/StateStore.java#L53) here.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        try {
+            log.info(String.format("Storing framework id: %s", fwkId));
+            storeFrameworkId(fwkId);
+        } catch (StateStoreException ex) {
+            log.error("Failed to set FrameworkID: " + fwkId, ex);
+            throw ex;
+        }
+    }
+
+    public void recordTasks(List<TaskInfo> taskInfos) throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: This should be transactional.  Maybe not best as part of this PR, though.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+            taskStatuses.add(taskStatus);
+        }
+
+        storeTasks(taskInfos);
+        for (TaskStatus taskStatus : taskStatuses) {
+            recordTaskStatus(taskStatus);
+        }
+    }
+
+    public void updateStatus(TaskStatus taskStatus) throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: What's the value of wrapping this?

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+        log.info(String.format("Recording updated TaskStatus to state store: %s", taskStatus));
+        recordTaskStatus(taskStatus);
+    }
+
+    public List<TaskInfo> getTerminatedTaskInfos() throws Exception {
+        return new ArrayList(fetchTerminatedTasks());
+    }
+
+    public int getRunningBrokersCount() throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Not relevant to dcos-commons?

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public Set<TaskStatus> getTaskStatuses() throws StateStoreException {
+        Set<TaskStatus> taskStatuses = new HashSet<TaskStatus>();
+        taskStatuses.addAll(fetchStatuses());
+        return taskStatuses;
+    }
+
+    public List<TaskInfo> getTaskInfos() throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Why wrap this?

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+        List<TaskInfo> taskInfos = new ArrayList<TaskInfo>();
+        taskInfos.addAll(fetchTasks());
+        return taskInfos;
+    }
+
+    public List<Resource> getExpectedResources() {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: [Already implemented here](https://github.com/mesosphere/dcos-commons/blob/6a5e665c56dcb9eb478b89619453f4f5e0fdbad2/src/main/java/org/apache/mesos/offer/ResourceCleaner.java#L103).  Should be factored out of both of these.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+                }
+            }
+        } catch (Exception ex) {
+            log.error("Failed to retrieve all Task information", ex);
+            return resources;
+        }
+        return resources;
+    }
+
+    public void recordTaskInfo(TaskInfo taskInfo) throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Not much value in wrapping this.

File: src/main/java/org/apache/mesos/offer/ResourceUtils.java (hunk 2 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/offer/ResourceUtils.java]: whoops, yea

File: src/main/java/org/apache/mesos/curator/CuratorStateStore.java (hunk 1 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/curator/CuratorStateStore.java]: I'll create an AbstractStateStore with these methods, so all implementations can share the functionality.

File: src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java (hunk 4 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java]: Whoops, this is from my debugging.  Good catch.  Removed.

File: src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java (hunk 5 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java]: whoops, removed.

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 6 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: changed

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 7 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: fixed

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 10 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: whoops, removed.

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 11 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: I just copy+pasted this from the previous `FrameworkState` class in kafka.  I can remove it and we can just use `Collection` if you like.

File: src/main/java/org/apache/mesos/curator/CuratorStateStore.java (hunk 1 above)
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/curator/CuratorStateStore.java]: Please use composition not inheritance.

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 12 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: removed

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 13 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Well, since I'm moving from inheritance to composition, we have to wrap this in order to expose it, unless you want me to expose the underlying `StateStore`

**Comment by mgummelt** [comment]
@gabrielhartmann ready for re-review

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+
+    public SchedulerState(AbstractStateStore stateStore) {
+        this.stateStore = stateStore;
+    }
+
+    public AbstractStateStore getStateStore() {
+        return stateStore;
+    }
+
+    public Optional<FrameworkID> getFrameworkId() {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Is the added value here the removal of the `throws StateStoreException` from the method signature?  I don't understand why this is needed or wanted.  [The CuratorStateStore in particular already performs the appropriate Exception handling and logging](https://github.com/mesosphere/dcos-commons/blob/master/src/main/java/org/apache/mesos/curator/CuratorStateStore.java#L140-L156).

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+        try {
+            return stateStore.fetchFrameworkId();
+        } catch (StateStoreException ex) {
+            log.warn("Failed to get FrameworkID. "
+                    + "This is expected when the service is starting for the first time.", ex);
+        }
+        return Optional.empty();
+    }
+
+    public void setFrameworkId(FrameworkID fwkId) throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Similar comment here, as to above.  These should be straightforward pass throughs to the `CuratorStateStore`.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+            recordTaskStatus(taskStatus);
+        }
+    }
+
+    public void updateStatus(TaskStatus taskStatus) throws StateStoreException {
+        log.info(String.format("Recording updated TaskStatus to state store: %s", taskStatus));
+        recordTaskStatus(taskStatus);
+    }
+
+    public List<TaskInfo> getTerminatedTaskInfos() throws Exception {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: This only converts a `Collection` to a `List`

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+        return new ArrayList(stateStore.fetchTerminatedTasks());
+    }
+
+    @Override
+    public Set<TaskStatus> getTaskStatuses() throws StateStoreException {
+        Set<TaskStatus> taskStatuses = new HashSet<TaskStatus>();
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: `Collection` to `Set` transformation, not necessary.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+        taskStatuses.addAll(stateStore.fetchStatuses());
+        return taskStatuses;
+    }
+
+    public List<TaskInfo> getTaskInfos() throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: `Collection` to `List` transformation.

File: src/main/java/org/apache/mesos/state/AbstractStateStore.java [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+
+/**
+ * Abstract StateStore implementation.
+ */
+public abstract class AbstractStateStore implements StateStore {
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/AbstractStateStore.java]: These feel like they could be static utility methods, rather than putting them in statestore proper

File: src/main/java/org/apache/mesos/state/AbstractStateStore.java [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    public void storePropertyAsObj(final String key, final Object obj) throws StateStoreException {
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        ObjectOutputStream oos = null;
+        try {
+            oos = new ObjectOutputStream(bos);
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/AbstractStateStore.java]: Will there be compatibility issues if we change the JDK version in the future?

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState implements TaskStatusProvider {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: I haven't gone through ever method, but I don't see this class adding a lot value, but it's adding some complexity, and doesn't really implement a clear interface, but selectively wraps a lot of what `StateStore` is doing.

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+    private static final String SUPPRESSED_KEY = "suppressed";
+
+    protected final AbstractStateStore stateStore;
+
+    public SchedulerState(String frameworkName, String mesosZkURI) {
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: It feels odd pulling in one of the concrete state store types here. Maybe just omit this constructor in favor of the following one?

File: src/main/java/org/apache/mesos/state/AbstractStateStore.java [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.io.*;
+
+/**
+ * Abstract StateStore implementation.
+ */
+public abstract class AbstractStateStore implements StateStore {
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    public void storePropertyAsObj(final String key, final Object obj) throws StateStoreException {
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/AbstractStateStore.java]: These could both be static methods of StateStore and avoid the definition of an abstract class, of which there are none (to my knowledge) in dcos-commons.

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 21 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: It looks like a lot of functions in this class are just performing datatype conversions against the underlying StateStore. Are those really needed or could they be omitted here?

File: src/main/java/org/apache/mesos/state/AbstractStateStore.java (hunk 20 above)
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/AbstractStateStore.java]: If this is just to transform boolean (isSuppressed) to byte array and back.  We could defer this complexity with something like:

```
boolean vIn = true;
byte vOut = (byte)(vIn?1:0);
```

File: src/main/java/org/apache/mesos/state/StateStoreUtils.java [hunk 24]
```diff
@@ -1,13 +1,16 @@
 package org.apache.mesos.state;
 
 import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Utilities for implementations and users of {@link StateStore}.
  */
 public class StateStoreUtils {
```
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/StateStoreUtils.java]: This seems to create an inconsistency in how scheduler state is stored:

If I want to set the frameworkID:

```
frameworkState.setFrameworkId(id)
```

But if I want to set the suppressed state:

```
StateStoreUtils.setSuppressed(frameworkState.getStateStore(), false);
```

These are both just two different properties of the scheduler, but now they have significantly different interfaces.  I'd rather just write:

```
frameworkState.setFrameworkId(id)
frameworkState.setSuppressed(false)
```

Can we instead make a concrete super class: `class SchedulerState` that implements `setSuppressed`, `getSuppressed`, `setFrameworkId`, etc?

**Comment by mgummelt** [comment]
@gabrielhartmann ready for re-review

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    private static final String SUPPRESSED_KEY = "suppressed";
+    private final StateStore stateStore;
+
+    public SchedulerState(StateStore stateStore) {
+        this.stateStore = stateStore;
+    }
+
+    public Optional<FrameworkID> getFrameworkId() {
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: This and setFrameworkId() feel unrelated to the rest of the class. Could the caller just invoke the stateStore functions directly?

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.mesos.Protos.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState  {
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Double space

File: src/main/java/org/apache/mesos/state/SchedulerState.java [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+        if (suppressed.length != 1) {
+            log.error("Unexpected SUPPRESSED byte array length: " + suppressed.length);
+            return false;
+        }
+
+        return (suppressed[0] == 1) ? true : false;
+    }
+
+    public void setSuppressed(boolean isSuppressed) {
+        final byte[] suppressed = new byte[]{(byte) (isSuppressed ? 1 : 0)};
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: Could just store a 'true' or 'false' ascii string. Then it'd be easy to diagnose when accessing the data directly (eg in exhibitor)

File: src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java [hunk 28]
```diff
@@ -41,7 +41,7 @@
  * as failed
  */
 public class RecoverySchedulerTest {
-    private DefaultRecoveryScheduler recoveryScheduler;
+    private DefaultRecoveryScheduler repairScheduler;
```
Comment (nickbp) [inline src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java]: Any reason for the rename? the type is RecoveryScheduler..

File: src/test/java/org/apache/mesos/state/SchedulerStateTest.java [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Test
+    public void testGetFrameworkId() {
+        schedulerState.setFrameworkId(TestConstants.frameworkId);
+        assertEquals(schedulerState.getFrameworkId().get(), TestConstants.frameworkId);
+    }
+
+    @Test
+    public void testIsSuppressed() {
+        schedulerState.setSuppressed(true);
```
Comment (nickbp) [inline src/test/java/org/apache/mesos/state/SchedulerStateTest.java]: Maybe add a call before setSuppressed() has been called, to test the behavior when `suppressed` is unset

File: src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java (hunk 28 above)
Comment (gabrielhartmann) [inline src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java]: This is just a mistake from before.  It used to be called a `DefaultRepairScheduler` but the variable name didn't keep up.

Comment (gabrielhartmann) [inline src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java]: Also, I'm confused.  I thought [I updated these already](https://github.com/mesosphere/dcos-commons/blob/master/src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java#L44).  @mgummelt: What's going on with the diff here?  Regardless, let's leave changes irrelevant to the content of this PR to other PRs.

**Comment by gabrielhartmann** [comment]
@mgummelt:  In general, I agree with @nickbp 's comments.

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 25 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: I will, sure.  

But how I really think this should be designed (possibly a future change) is:
- StateStore should be a lower level interface that just deals with key/value storage.
- SchedulerState should have all the scheduler specific state (frameworkId, suppressed, task state, etc.)

As it is, some scheduler state is exposed in StateStore, and some is exposed in SchedulerState (or equivalently, KafkaState and CassandraState), and the distinction seems arbitrary.

Thoughts?

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 27 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: The 0 and 1 are visible in exhibitor, and given that they're more canonical binary representations of a boolean, I'd rather keep it as is.

File: src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java (hunk 28 above)
Comment (mgummelt) [inline src/test/java/org/apache/mesos/scheduler/recovery/RecoverySchedulerTest.java]: Yea, not sure what happened here.  Will fix.

File: src/main/java/org/apache/mesos/state/SchedulerState.java (hunk 26 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: what do you mean?

Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/SchedulerState.java]: fixed

**Comment by mgummelt** [comment]
@nickbp @gabrielhartmann ready for re-review
